### PR TITLE
Resolves #17543 TTS availability among devices

### LINKED ISSUE #17543: TTS: Better handle the case where no Engines are available
On an Amazon Fire Stick, there are no default TTS Engines.

We should figure out how we want to handle the UX of this, and inform the user of where to look to resolve the issue

Ref: https://www.reddit.com/r/Anki/comments/1h67g6c/tts_on_anki_for_amazon_firestick/

### PR DESCRIPTION
**Pull request by Ssdosaofc**
Resolves #17543 Added Text-to-Speech engine availability check, accounting for devices like Amazon Fire Stick.
Well, here I might need some help implementing all string values for "tts_unavailable_title" and "tts_unavailable_message." :)

### CODE REVIEW

**Comment by lukstbit** [comment]
Hey, this has commits from your previous PR and makes it harder to see what are the real changes. You should _git rebase_ on top of main to only see here the commit with the actual changes.

**Comment by Ssdosaofc** [comment]
> Hey, this has commits from your previous PR which makes it harder to see what are the real changes. You should _git rebase_ on top of main to only see here the commit with the actual changes.

I am really sorry. I am new to all this. I will take care of this in the future.

**Comment by Ssdosaofc** [comment]
Thank you for pointing it out. Now it looks fine.

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AdvancedSettingsFragment.kt [hunk 1]
```diff
@@ -76,6 +77,20 @@ class AdvancedSettingsFragment : SettingsFragment() {
...
+            if (!ttsAvailable) {
+                AlertDialog.Builder(requireContext()).show {
+                    setTitle(R.string.tts_unavailable_title)
+                    setMessage(R.string.tts_unavailable_message)
+                    setIcon(R.drawable.ic_warning)
+                    setPositiveButton(R.string.dialog_ok) { _, _ -> }
+                }
+                ttsPref.isChecked = false
+                return@setOnPreferenceChangeListener false
+            }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/AdvancedSettingsFragment.kt:92]: This feature is intended to be deleted.

We need to handle the unified mechanism of TTS:

https://docs.ankiweb.net/templates/fields.html#text-to-speech-for-individual-fields

https://github.com/ankidroid/Anki-Android/blob/d8dcf7e983cf0aaca232d687da6ff8ca024a0eeb/AnkiDroid/src/main/java/com/ichi2/libanki/Tts.kt#L45

**Pull request closed without merging**